### PR TITLE
HCAL 2017 HF TP dual-anode readout fixes.

### PR DESCRIPTION
**Pull request by matz-e**
Fix the energy calculation to match firmware. Duplicate valid energies when one part of the readout fails, and shift over the lower energy bits properly.

### CODE REVIEW

**Comment by kpedro88** [comment]
@cmsbuild please test

File: SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc [hunk 1]
```diff
@@ -595,16 +595,17 @@ void HcalTriggerPrimitiveAlgo::analyzeHF2017(
...
+               if (long_fiber_count == 1)
+                  long_fiber_val *= 2;
+               if (short_fiber_count == 1)
+                  short_fiber_val *= 2;
 
                auto sum = long_fiber_val + short_fiber_val;
                // If both towers are valid, we cut the sum in half
-               if (long_fiber_count > 0 and short_fiber_count > 0)
-                  sum *= 0.5;
+               if (long_fiber_count == 0 or short_fiber_count == 0)
```
Comment (davidlange6) [inline SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc]: hi @matz-e - this logic is very cryptic for me - is there a more concise (and less-codey) summary we can refer to?

Comment (matz-e) [inline SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc]: I should update the comment (was focused on logic) to read
```c++
// If one of the fibers is invalid, use the value of the other one instead by doubling the sum
```
Other than that, I can also link to the uHTR spec, which is a bit more verbose and covers a lot more.

Would that be better?

Comment (kpedro88) [inline SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc]: @davidlange6 is that comment acceptable?

**Comment by rekovic** [comment]
+1

File: SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc (hunk 1 above)
Comment (matz-e) [inline SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc]: @davidlange6 Just updated this with a fairly (IMO) verbose comment on the treatment of the energy calculations.

**Pull request merged**